lib/rmt/scc.rb

```diff
@@ -86,16 +86,21 @@ def sync_systems
     # do not sync BYOS proxy systems to SCC
     systems = System.where('scc_registered_at IS NULL OR last_seen_at > scc_registered_at').where(proxy_byos: false)
     @logger.info(_('Syncing %{count} updated system(s) to SCC') % { count: systems.size })
-    scc_api_client.send_bulk_system_update(systems) do |successful_response|
-      next if successful_response[:systems].count == 0
-
-      successful_response[:systems].each do |system_hash|
-        # Update attributes without triggering after_update callback (which resets scc_synced_at to nil)
-        System.find_by(login: system_hash[:login])
-           .update_columns(scc_system_id: system_hash[:id], scc_synced_at: Time.current)
-      rescue StandardError => e
-        @logger.error(_('Failed to sync systems: %{error}') % { error: e.to_s })
-      end
+
+    updated_systems = scc_api_client.send_bulk_system_update(systems)
+    if updated_systems[:systems].blank?
+      # Case: silently fail but respond with 201
+      systems.update_all(scc_registered_at: Time.current)
```

Comment:
This will mark the failed systems as synced now and will try again on the next run. The sync issues are due to unique indexes on login-pass-token. We can keep track of failed syncs but that is outside of this bug fix.

Reply:
I think it will not try again to sync them on the next run, because they have an updated `scc_registered_at` timestamp, so they will get skipped. 
Not updating `scc_registered_at` would include them again in the next sync run.

Reply:
Well, in that case then there's no bug. Because if the systems sync silently fail (scc returns 2xx), we don't do anything. Rather this is a feature

Reply:
The bug we need to fix lays in the code below (line 94 in master): It only did a `find_by(login: ...` which in case there are system clones with the same login, only selected and updated the sync timestamp of the first one.

Reply:
I will add a test case tomorrow for it

Reply:
> Explanation
> The system sync service is randomized every 9hrs. This would ensure that scc_registered_at will always be greater that last_seen_at (scc needs a diff of 3 mins) - check [here](https://github.com/SUSE/rmt/blob/6da777085d3e11b287c4ca4c04c52fc0ace65625/lib/rmt/scc.rb#L87). Hence, during the next systemd timer run, the failed system will be included.

But we only include systems in the sync that have `last_seen_at > scc_registered_at`, so with what you described above, the system will _not_ get included in the sync.